### ISSUE TITLE
Trainer save best model

### ISSUE THREAD

**Issue by Andrewwango** [deepinv/deepinv#398]
Currently, `deepinv.Trainer` only has the option to save the model every n epochs. It would be good to have functionality, e.g. `ckp_interval='best'`, where the model gets saved every time if the eval metric is greater than the current best.

**Comment by alexmehta** [comment]
Would a good solution be to implement ability to add hooks/callbacks when training like keras lightning and other training libraries do? Would allow for saving best model but also might make other features such as early stopping, wandb/tensorboard integration more clean/extensible.

**Comment by Andrewwango** [comment]
Thanks for the ideas @alexmehta !

I think an eventual solution that might be cleaner and more scalable than what we have now could be achieved with hooks/callbacks. Currently our framework is to provide modular methods with one unified set of class params that are passed around.

I am still unsure whether callbacks are appropriate right now, so for now we have added this feature in #437 in our existing framework, but I will add your comment to discussions as I think it will be likely that we will need your advice soon again!
